inc/Utf8/PhpString.php

```diff
@@ -27,9 +27,8 @@ public static function basename($path, $suffix = '')
             $path = substr($path, $rpos + 1);
         }
 
-        $suflen = strlen($suffix);
-        if ($suflen && (substr($path, -$suflen) === $suffix)) {
-            $path = substr($path, 0, -$suflen);
+        if (str_ends_with($path, $suffix)) {
+            $path = substr($path, 0, -strlen($suffix));
```

Comment:
Empty suffix results in str_end_with == true, does that make a difference here?

Reply:
I have looked at this as well. At first glance I don't see any effective difference. But if `$suffix === ''` then the statement `$path = substr($path, 0, -strlen($suffix));` becomes: `$path = substr($path, 0, -0);` and `-0 === 0` so yeah, that will make a difference.
Fixed.

Reply:
This solved also all issues with the unit tests :-)